packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1023,11 +1033,19 @@ function validateTemplate(
 
     // At this point in the parsing all supported attributes from a non root template
     // should have been validated and removed from ParsedAttribute.
-    const templateAttrs = parsedAttr.getAttributes();
-    if (templateAttrs.length) {
-        ctx.warnAtLocation(ParserDiagnostics.INVALID_TEMPLATE_ATTRIBUTE, location, [
-            templateAttrs.map((attr) => attr.name).join(', '),
-        ]);
+    const invalidTemplateAttributes = parsedAttr.getAttributes();
+    if (invalidTemplateAttributes.length) {
+        // Warn when there are valid template attributes attached to the template
+        if (validTemplateAttributes) {
+            ctx.warnAtLocation(ParserDiagnostics.INVALID_TEMPLATE_ATTRIBUTE_WARNING, location, [
+                invalidTemplateAttributes.map((attr) => attr.name).join(', '),
+            ]);
+        } else {
+            // Throw an error when there are no valid attributes attached to the template
+            ctx.throwAtLocation(ParserDiagnostics.INVALID_TEMPLATE_ATTRIBUTE, location, [
+                invalidTemplateAttributes.map((attr) => attr.name).join(', '),
+            ]);
+        }
```

Comment:
Plan to gate the error behind a flag, pushed this up, for now, to see how many downstream dependencies are affected.